docs/index.rst

```diff
@@ -10,51 +10,102 @@
+grand canonical ensemble, PyTNG does not currently support this. Additonally,
+the TNG format includes a TNG_MOLECULES block that contains the simulation
+topology. PyTNG does not currently make use of this information.
 
-TNG files can be read using the TNGFile class as a file handle,
+
+Usage example for TNGFileIterator
+=================================
+
+TNG files can be read using the TNGFileIterator class as a file handle,
 which supports use as a context manager.
-The TNGFile returns one frame at a time, which each frame being returned as a
-namedtuple with the following attributes:
 
-============== ======== =======================================
-attribute      type     description
-============== ======== =======================================
-``positions``  float32  coordinates for each atom in the frame
-``time``       float    current system time
-``step``       int      frame index
-``box``        float32  3x3 matrix of the system volume
-============== ======== =======================================
+The TNGFileIterator has attributes related to the trajectory metadata, such as
+the number of integrator steps, the number of steps with data, the block_ids 
+available at each step, and the stride at which each block is written.
+
+The TNGFileIterator returns one frame at a time, which is accessed from the
+:attr:`TNGFileIterator.current_integrator_step` attribute or as part of any
+slicing or indexing operation. A NumPy array of the right size must then be
+provided to a getter method for the data to be read into.
+
+An example of how to read positions from a TNG file is shown below.
 
-For example, the coordinate information can be accessed via the `.positions` 
-attribute of the returned object.
 
 .. code-block:: python
 
-  import pytng
+   import pytng
+   import numpy as np
+
+   with pytng.TNGFileIterator("traj.tng", 'r') as tng:
+
+      # make a numpy array to hold the data using helper function
+      # this array will then be updated in-place 
+      
+      positions = tng.make_ndarray_for_block_from_name("TNG_TRAJ_POSITIONS")
+      
+      # the TNG API uses regular strides for data deposition, here we stride
+      # over the whole trajectory for the frames that have position data
+      # len(tng) is the total number of steps in the file.
+
+      for ts in tng[0:len(tng):tng.block_strides["TNG_TRAJ_POSITIONS"]]:
+         # read the integrator timestep, modifying the current_integrator_step
+         # which is returned as ts
+
+         # get the data from the requested block by supplying NumPy array which
+         # is updated in-place or returned 
+         
+         positons = ts.get_positions(positions)
 
-  with pytng.TNGFile('traj.tng', 'r') as f:
-      for ts in f:
-          ts.positions
+           
 
 It is also possible to slice and index the file object to select particular
 frames
 
 .. code-block:: python
 
   import pytng
+  import numpy as np
```

Comment:
Why numpy here? You're not using it (and presumably assume that `positions` is already allocated)